src/garage/np/algos/cma_es.py

```diff
@@ -40,16 +40,22 @@ def __init__(self,
 
         self.sigma0 = sigma0
 
+    def sample_params(self):
+        return self.es.ask(self.n_samples)
+
+    def _initialize_vars(self):
+        """Initialize variables before training."""
         init_mean = self.policy.get_param_values()
-        self.es = cma.CMAEvolutionStrategy(init_mean, sigma0)
+        self.es = cma.CMAEvolutionStrategy(init_mean, self.sigma0)
 
         self.all_params = self.sample_params()
         self.cur_params = self.all_params[0]
         self.policy.set_param_values(self.cur_params)
         self.all_returns = []
 
-    def sample_params(self):
-        return self.es.ask(self.n_samples)
+    def train(self, runner, batch_size):
+        self._initialize_vars()
```

Comment:
I suspect doing so will fail resume mechanism.

Reply:
This is actually fine. Please take a look at what the initialization is saying. It is not initializing tf tensors' value. @naeioi

Reply:
I think to avoid confusion. Let's just put the content of `_initialize_vars` to `train` or rename this method. I am pretty bad at naming stuff so please give it a good name :).

Reply:
If naming is the problem, we can just put the content inside. (Being said, I guess `_initialize_vars` is good here unless someone can figure out a better one)

Reply:
Ah true. It's me who wrote this function and I even forget how it works! Thanks for pointing out @zhanpenghe .